akka-typed/src/main/scala/akka/typed/cluster/Cluster.scala

```diff
@@ -0,0 +1,314 @@
+ *
+ * If all you are interested in is to react on this node becoming a part of the cluster,
+ * see [[OnSelfUp]]
+ *
+ * @param subscriber A subscriber that will receive events until it is unsubscribed or stops
+ * @param eventClass The type of events to subscribe to, can be individual event types such as
+ *                   `ReachabilityEvent` or one of the common supertypes, such as `MemberEvent` to get
+ *                   all the subtypes of events.
+ */
+final case class Subscribe[A <: ClusterDomainEvent](
+  subscriber: ActorRef[A],
+  eventClass: Class[A]) extends ClusterStateSubscription
+
+/**
+ * Subscribe to this node being up, after sending this event the subscription is automatically
+ * cancelled. If the node is already up the event is also sent to the subscriber. If the node was up
+ * but is no more because it left or is leaving the cluster, no event is sent and the subscription
+ * request is ignored.
+ */
+final case class OnSelfUp(subscriber: ActorRef[SelfUp]) extends ClusterStateSubscription
+
+/**
+ * @param currentClusterState The cluster state snapshot from when the node became Up.
+ */
+final case class SelfUp(currentClusterState: CurrentClusterState)
+
+/**
+ * Subscribe to this node being removed from the cluster. If the node was already removed from the cluster
+ * when this subscription is created it will be responded to immediately from the subscriptions actor.
+ */
+final case class OnSelfRemoved(subscriber: ActorRef[SelfRemoved.type]) extends ClusterStateSubscription
+
+// TODO any additional data worth passing with this response?
+final case object SelfRemoved
+
+final case class Unsubscribe[T](subscriber: ActorRef[T]) extends ClusterStateSubscription
+final case class GetCurrentState(recipient: ActorRef[CurrentClusterState]) extends ClusterStateSubscription
+
+/**
+ * Not intended for user extension.
+ */
+@DoNotInherit
+sealed trait ClusterCommand
+
+/**
+ * Try to join this cluster node with the node specified by 'address'.
+ *
+ * An actor system can only join a cluster once. Additional attempts will be ignored.
+ * When it has successfully joined it must be restarted to be able to join another
+ * cluster or to join the same cluster again.
+ *
+ * The name of the [[akka.actor.ActorSystem]] must be the same for all members of a
+ * cluster.
+ */
+final case class Join(address: Address) extends ClusterCommand
+
+/**
+ * Scala API:
+ * Join the specified seed nodes without defining them in config.
+ * Especially useful from tests when Addresses are unknown before startup time.
+ *
+ * An actor system can only join a cluster once. Additional attempts will be ignored.
+ * When it has successfully joined it must be restarted to be able to join another
+ * cluster or to join the same cluster again.
+ */
+final case class JoinSeedNodes(seedNodes: immutable.Seq[Address]) extends ClusterCommand {
+
+  /**
+   * Java API:
+   *
+   * Join the specified seed nodes without defining them in config.
```

Comment:
I think the javadoc looks better if this is immediately after `Java API: `. First sentence.